src/script/event/EventRepository.ts

```diff
@@ -572,7 +572,8 @@ export class EventRepository {
     }
 
     const containsLinkPreview = newEventData.previews && !!newEventData.previews.length;
-    const isRetryAttempt = originalEvent.status === StatusType.FAILED;
+    const isRetryAttempt =
+      originalEvent.status === StatusType.FAILED || originalEvent.status === StatusType.FEDERATION_ERROR;
```

Comment:
This could be a typeguard function to be able to reuse this check aka
isFailed(event) and isFederationError(event)